packages/falso/src/lib/credit-card-number.ts

```diff
@@ -12,19 +27,42 @@ import { randNumber } from './number';
  *
  * @example
  *
+ * randCreditCardNumber({ issuer: 'Visa'})
+ *
+ * @example
+ *
  * randCreditCardNumber({ length: 10 })
  *
  */
-export function randCreditCardNumber<Options extends FakeOptions>(
-  options?: Options
-) {
-  return fake(() => {
-    let cardNumber = '';
-
-    for (let i = 0; i < randNumber({ min: 12, max: 19 }); i++) {
-      cardNumber += randNumber({ min: 0, max: 9 });
-    }
-
-    return cardNumber;
-  }, options);
+
+export function randCreditCardNumber<
+  Options extends FakeOptions & {
+    issuer?: Issuer;
+  }
+>(options?: Options) {
+  let card = rand(data);
```

Comment:
I'm not sure why you changed the implementation. Each function should use the `fake` function.

Reply:
Sorry I misunderstood your previous comment. You are telling to skip `rand` function, right?

Reply:
Anyways I need to a random value from data to get a format from any one provider, that's why I did that. Do you think we no need to do that?